### ISSUE TITLE
Pagefind for search

### ISSUE THREAD

**Issue by corneliusroemer** [loculus-project/loculus#3079]
Docs and governance pages are hard to find (there are more than 30 or so in Pathoplexus).

We could use [Pagefind](https://pagefind.app/) to implement a site-wide search, as discussed in https://github.com/pathoplexus/pathoplexus/issues/244

I looked into this briefly:
- We can't use the astro-pagefind plugin because we use server side rendering.
- Instead, we can scrape pages with, for instance, `wget` to create a folder tree with render html pages. E.g.:
  ```
  npm run start-server &
  wget --mirror \
       --convert-links \
       --adjust-extension \
       --no-parent \
       --reject "*.js,*.css,*.png,*.jpg,*.jpeg,*.gif,*.ico,*.svg,*.woff,*.woff2,*.ttf" \
       --exclude-directories="/seq" \
       --directory-prefix=./static \
       localhost:3000
  ```
- We then run `npx pagefind --site static/localhost:3000`
- And move the resulting output into the public folder: `mv static/localhost:3000/pagefind public/pagefind`
- We can then add search in the header component of every page by:
  - including this in the html header:
    ```html
    <link href='/pagefind/pagefind-ui.css' rel='stylesheet' />
    <script src='/pagefind/pagefind-ui.js' is:inline></script>
    ```
  - Adding a search div somewhere in the header nav html:
    ```html
    <div id='search' class='px-4'></div>
    ```
  - Initializing the search somewhere:
    ```html
    <script is:inline>
        window.addEventListener('DOMContentLoaded', (event) => {
            new PagefindUI({
                element: '#search',
                showSubResults: true,
            });
        });
    </script>
    ```
- Here's an (ugly) proof-of-concept:
![2024-10-27 02 11 18](https://github.com/user-attachments/assets/0437168a-6b69-4f3c-916c-098fdb6b9e7e)

**Comment by corneliusroemer** [comment]
We could also use Algolia, might be slightly more feature packed, has free plan and if we need more we should qualify for open source: https://www.algolia.com/for-open-source/

**Comment by theosanderson** [comment]
I don't think we should incorporate something into Loculus that requires filling out a form to apply(?)
